src/Phinx/Db/Adapter/PdoAdapter.php

```diff
@@ -389,7 +389,7 @@ public function getVersionLog(): array
         }
 
         foreach ($rows as $version) {
-            $result[(int)$version['version']] = $version;
+            $result[$version['version']] = $version;
```

Comment:
I'm not totally convinced this won't introduce errors where PHP will automatically cast integer keys to integers, according to `PHP_INT_MAX`, e.g. on PHP-32bit:

```
php > print(PHP_INT_MAX);
2147483647
php > $a = [];
php > $a["2147483647"] = 1;
php > var_dump($a);
array(1) {
  [2147483647]=>
  int(1)
}
php > $a["21474836479999"] = 1;
php > var_dump($a);
array(2) {
  [2147483647]=>
  int(1)
  ["21474836479999"]=>
  int(1)
}
```

and that I'm worried we end up with potential issues around sorting when we use the key.

If we wanted to support PHP 32-bit, my preference would be to make a breaking change and remove the usage of the version as an array index everywhere, and then when we need to do any type of sorting, we do any sorts over the values (be it the version string or a migration object we pull the version off of).

Reply:
I agree, this would be better. Let me investigate in that direction.